modules/Oculus_module/src/RobotControlHelper.cpp

```diff
@@ -0,0 +1,300 @@
+    }
+    return true;
+}
+
+bool RobotControlHelper::setDirectPositionReferences(const yarp::sig::Vector& desiredPosition)
+{
+    if (!m_isActive)
+        return true;
+
+    if (m_positionDirectInterface == nullptr)
+    {
+        yError()
+            << "[RobotControlHelper::setDirectPositionReferences] PositionDirect I/F not ready.";
+        return false;
+    }
+
+    if (desiredPosition.size() != m_actuatedDOFs)
+    {
+        yError() << "[RobotControlHelper::setDirectPositionReferences] Dimension mismatch between "
+                    "desired position vector and the number of controlled joints.";
+        return false;
+    }
+
+    double threshold = 5;
+    for (int i = 0; i < m_actuatedDOFs; i++)
+    {
+        m_desiredPositionInDegrees(i) = iDynTree::rad2deg(desiredPosition(i));
+
+        // check if the position is outside the limit
+        if (m_desiredPositionInDegrees(i) <= m_minJointsPosition(i))
+            m_desiredPositionInDegrees(i) = m_minJointsPosition(i) + threshold;
+        else if (m_desiredPositionInDegrees(i) >= m_maxJointsPosition(i))
+            m_desiredPositionInDegrees(i) = m_maxJointsPosition(i) - threshold;
+    }
+
+    // set position diret mode
+    if (!m_positionDirectInterface->setPositions(m_desiredPositionInDegrees.data()))
+    {
+        yError() << "[RobotControlHelper::setDirectPositionReferences] Error while setting the "
+                    "desired position.";
+        return false;
+    }
+
+    return true;
+}
+
+void RobotControlHelper::updateTimeStamp()
+{
+    if (!m_isActive)
+        return;
+
+    yInfo() << "ciao";
+
+    if (m_timedInterface)
+        m_timeStamp = m_timedInterface->getLastInputStamp();
+    else
+        m_timeStamp.update();
+}
+
+bool RobotControlHelper::getFeedback()
+{
+    if (!m_isActive)
+        return true;
+
+    yInfo() << "ciao";
+
+    if (!m_encodersInterface->getEncoders(m_positionFeedbackInDegrees.data()))
+    {
+        yError() << "[RobotControlHelper::getFeedbacks] Unable to get joint position";
+        return false;
+    }
+
+    for (unsigned j = 0; j < m_actuatedDOFs; ++j)
+        m_positionFeedbackInRadians(j) = iDynTree::deg2rad(m_positionFeedbackInDegrees(j));
+
+    return true;
+}
+
+const yarp::os::Stamp& RobotControlHelper::timeStamp() const
+{
+    return m_timeStamp;
+}
+
+yarp::os::Stamp& RobotControlHelper::timeStamp()
+{
+    return m_timeStamp;
+}
+
+const yarp::sig::Vector& RobotControlHelper::jointEncoders() const
+{
+    return m_positionFeedbackInRadians;
+}
+
+void RobotControlHelper::close()
+{
+    if (!switchToControlMode(VOCAB_CM_POSITION))
+    {
+        yError() << "[close] Unable to switch in position control.";
+    }
+
+    if (!m_robotDevice.close())
+        yError() << "[close] Unable to close the device.";
+}
+
+int RobotControlHelper::getDoFs()
+{
+    return m_actuatedDOFs;
+}
+
+yarp::sig::Matrix RobotControlHelper::getLimits()
+{
+    yarp::sig::Matrix limits(m_minJointsPosition.size(), 2);
```

Comment:
I would read the joint limits here. In case it is not possible to retrieve a limit, simply set it equal to zero on both sides. Eventually you can also output a vector of bools specifying which joint is in a "not configured" status.

Reply:
Instead of zero, the method may set the limits equal to the current encoder value

Reply:
Done in 185a6dd